flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/ScalarFunction.scala

```diff
@@ -58,6 +60,11 @@ abstract class ScalarFunction extends UserDefinedFunction {
 
   override def toString: String = getClass.getCanonicalName
 
+  final def functionIdentifier: String = {
```

Comment:
In order to keep JAVA's interface and SCALAR's trait consistent, I prefer not to do any implementation in UserDefinedFunction.